site-scrapers/HeywoodHealthcare/index.js

```diff
@@ -0,0 +1,25 @@
+const { site } = require("./config");
+
+const noAppointmentMatchString =
+    "All appointment types are private, none are available for scheduling.";
+
+module.exports = async function GetAvailableAppointments(browser) {
+    console.log(`${site.name} starting.`);
+    const webData = await ScrapeWebsiteData(browser);
+    console.log(`${site.name} done.`);
+    return webData;
+};
+
+async function ScrapeWebsiteData(browser) {
+    const page = await browser.newPage();
+    await page.goto(site.signUpLink);
+    // Is the clinic open?
+    const alertElement = await page.$("div.alert[data-qa=error-notice]");
```

Comment:
I think this will still fail (I'm guessing this div won't show up if there's availability, right?).
What we want is to do
```
await page.$("...").catch((error) => { return null; });
```
or something of that sort. When the query doesn't find an element with that selector I'm pretty sure it returns a rejected promise. We have to catch that or the script will bail.
A good way to test this without writing unit tests is to change that selector to gibberish - something you know doesn't exist on the page currently. Then you can test the behavior is as expected.

Reply:
@livgust I ended up writing three unit tests, one of which tests for the condition when the alert does not exist. They are included in the test suite and ran successfully through the GitHub workflow.